packages/block-library/src/embed/test/index.native.js

```diff
@@ -700,4 +723,208 @@ describe( 'Embed block', () => {
+			} )
+		);
+	} );
+
+	it( 'sets block caption', async () => {
+		const expectedCaption = 'Caption';
+
+		const waitForElement = ( { getByA11yLabel } ) =>
+			getByA11yLabel( /Embed Block\. Row 1/ );
+		const {
+			element,
+			getByPlaceholderText,
+			getByDisplayValue,
+		} = await initializeEditor(
+			{ initialHtml: RICH_TEXT_EMBED_HTML },
+			{ waitForElement }
+		);
+
+		// Select block
+		fireEvent.press( element );
+
+		// Set a caption
+		const captionField = getByPlaceholderText( 'Add caption' );
+		fireEvent( captionField, 'focus' );
+		fireEvent( captionField, 'onChange', {
+			nativeEvent: {
+				eventCount: 1,
+				target: undefined,
+				text: expectedCaption,
+			},
+		} );
+
+		// Get current caption
+		const caption = await waitFor( () =>
+			getByDisplayValue( `<p>${ expectedCaption }</p>` )
+		);
+
+		expect( caption ).toBeDefined();
+		expect( getEditorHtml() ).toMatchSnapshot();
+	} );
+
+	it( 'displays cannot embed on the placeholder if preview data is null', async () => {
+		// Return null response for requests to oembed endpoint.
+		fetchRequest.mockImplementation( ( { path } ) => {
+			const isEmbedRequest = path.startsWith( '/oembed/1.0/proxy' );
+			return Promise.resolve( isEmbedRequest ? EMBED_NULL_RESPONSE : {} );
+		} );
+
+		const initialHtml = RICH_TEXT_ERROR_EMBED_HTML;
+
+		const waitForElement = ( { getByA11yLabel } ) =>
+			getByA11yLabel( /Embed Block\. Row 1/ );
+		const { element, getByText } = await initializeEditor(
+			{
+				initialHtml,
+			},
+			{ waitForElement }
+		);
+
+		// Select block
+		fireEvent.press( element );
+
+		const cannotEmbedText = getByText( 'Unable to embed media' );
+
+		expect( cannotEmbedText ).toBeDefined();
+		expect( getEditorHtml() ).toMatchSnapshot();
+	} );
+
+	describe( 'block settings', () => {
+		it( 'toggles resize for smaller devices media settings', async () => {
+			const waitForElement = ( { getByA11yLabel } ) =>
+				getByA11yLabel( /Embed Block\. Row 1/ );
+			const {
+				element,
+				getByA11yLabel,
+				getByText,
+			} = await initializeEditor(
+				{ initialHtml: RICH_TEXT_EMBED_HTML },
+				{ waitForElement }
+			);
+
+			// Select block
+			fireEvent.press( element );
+
+			fireEvent.press(
+				await waitFor( () => getByA11yLabel( 'Open Settings' ) )
+			);
+
+			fireEvent.press(
+				await waitFor( () => getByText( /Resize for smaller devices/ ) )
+			);
+
+			expect( getEditorHtml() ).toMatchSnapshot();
+		} );
+
+		it( 'does not show settings button if responsive is not supported', async () => {
+			const waitForElement = ( { getByA11yLabel } ) =>
+				getByA11yLabel( /Embed Block\. Row 1/ );
+			const { element, getByA11yLabel } = await initializeEditor(
+				{ initialHtml: WP_EMBED_HTML },
+				{ waitForElement }
+			);
+
+			// Select block
+			fireEvent.press( element );
+
+			let settingsButton;
+			try {
+				settingsButton = await waitFor( () =>
+					getByA11yLabel( 'Open Settings' )
+				);
+			} catch ( e ) {
+				// NOOP
+			}
```

Comment:
This waits for the timeout of `waitFor` to be triggered, so it adds extra time to the test duration.